File: my-folder/0102-binary-tree-level-order-traversal/solution.java
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if (root == null) {
          return new ArrayList<>();
        }
    
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);
        
        List<List<Integer>> result = new ArrayList<>();
        
        while (!q.isEmpty()) {
            int qSize = q.size();
            
            List<Integer> levelList = new ArrayList<>();
            
            for (int i = 0; i < qSize; i++) {
               TreeNode polledNode = q.poll();
               
               levelList.add(polledNode.val);
               
               if (polledNode.left != null) {
                  q.offer(polledNode.left);
               }
               
               if (polledNode.right != null) {
                  q.offer(polledNode.right);
               }
            }
            
           result.add(levelList); 
        }
        
       return result; 
    }
}

File: my-folder/0102-binary-tree-level-order-traversal/solution.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
fun levelOrder(root: TreeNode?): List<List<Int>> {
    val output = ArrayList<List<Int>>()
    val queue = LinkedList<TreeNode>()

    return if (root == null) output
    else {
        queue.add(root)

        while (queue.isNotEmpty()) {
            val list = ArrayList<Int>()

            for (i in 0 until queue.size) {
                val polled = queue.poll()
                list.add(polled.`val`)
                
                
                polled.left?.let { queue.add(it) }
                polled.right?.let { queue.add(it) }
            }
            
            output.add(list)
        }

        output
    }
}
}
